### ISSUE TITLE
Motion planner

### ISSUE THREAD

**Issue by Barbany** [f3rm/f3rm#8]
In the paper, you mention that once you have a ranked list of poses, you use an inverse kinematics solver and a motion planner in PyBullet. However, your robot code does not include this. Could you please share the code for that or provide some implementation tips?

For instance, do you extract a single mesh from the learned NeRF, feed it to PyBullet along with the model of the robotic arm and gripper, and then plan motions for all the predicted grasps (from best to worst) until there is one that is feasible?

Thanks for your great work and your intention to share everything you've done with the community!"

**Comment by williamshen-nz** [comment]
> For instance, do you extract a single mesh from the learned NeRF

We tried extracting a mesh using marching cubes, and then running V-HACD on them so Pybullet could do efficient collision checking. However, we found that V-HACD was very slow and this degraded the quality of the geometry especially in fine details.

For collision checking, we checked against the sampled point cloud by adding them to the scene. There is definitely a better and faster way to do collision checking.

> feed it to PyBullet along with the model of the robotic arm and gripper, and then plan motions for all the predicted grasps (from best to worst) until there is one that is feasible?

Yes, that's right. We iterated through all the grasps, tried to solve IK and find a motion plan for each and executed the first one with a feasible path. We also had pre-grasp and post-grasp poses that were mostly just offsets in the z-axis of the end-effector.

Thanks!
